File: Algorithms/Game Theory/Alice and Bob's Silly Game.py
import os
import sys

#
# Complete the 'sillyGame' function below.
#
# The function is expected to return a STRING.
# The function accepts INTEGER n as parameter.
#

def sieve_of_eratosthenes(n):
    primes = [True] * (n + 1)
    p = 2
    while p * p <= n:
        if primes[p]:
            for i in range(p * p, n + 1, p):
                primes[i] = False
        p += 1
    prime_count = sum(1 for p in range(2, n + 1) if primes[p])
    return prime_count

def sillyGame(n):
    # Write your code here
    prime_count = sieve_of_eratosthenes(n)
    if prime_count % 2 == 1:
        return "Alice"
    else:
        return "Bob"


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    g = int(input().strip())

    for g_itr in range(g):
        n = int(input().strip())

        result = sillyGame(n)

        fptr.write(result + '\n')

    fptr.close()

File: Algorithms/Game Theory/Alice and Bob's Silly Game.php
<?php

/*
 * Complete the 'sillyGame' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER n as parameter.
 */

function sieve_of_eratosthenes($n) {
    $primes = array_fill(0, $n + 1, True);
    $p = 2;

    while ($p * $p <= $n) {
        if ($primes[$p] == True) {
            for ($i = $p * $p; $i <= $n; $i += $p) {
                $primes[$i] = False;
            }
        }
        $p++;
    }

    $prime_count = 0;
    for ($p = 2; $p <= $n; $p++) {
        if ($primes[$p] == True) {
            $prime_count++;
        }
    }

    return $prime_count;
}

function sillyGame($n)
{
    # Write your code here
    $prime_count = sieve_of_eratosthenes($n);

    if ($prime_count % 2 == 1) {
        return "Alice";
    } else {
        return "Bob";
    }
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$g = intval(trim(fgets(STDIN)));

for ($g_itr = 0; $g_itr < $g; $g_itr++) {
    $n = intval(trim(fgets(STDIN)));

    $result = sillyGame($n);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);
